docs/DeveloperGuide.md

```diff
@@ -1,38 +1,275 @@
+
+3. Open the command terminal and navigate to the folder where you downloaded `mealcompanion.jar` using `cd <file path>` commands as necessary.
+4. Ensure the terminal is within the correct folder.
+5. Use the command `java -jar mealcompanion.jar` to launch Meal Companion.
+![startup1.png](images/startup1.png)
+6. When you are finished using Meal Companion, use the `bye` command to terminate the application. This will ensure your stored data is saved correctly for future access.
+
+Please be advised that we cannot guarantee complete functionality of Meal Companion if you have modified the source code in your own local IDE. If you have any queries, feel free to reach out to any member of the development team and we will do our best to assist you.
+
+###### [Back to table of contents](#table-of-contents)
+
+## Design
+
+### Command Parsing
+
+The system of `Routable` classes is built to allow for a composable setup of commands and subcommands.
+
+Developers are able to build a "routing tree" by nesting `CommandRouterNode`s together, allowing for command matching
+to be done in a robust and extensible way. This architecture was chosen because our commands are designed to have common
+prefixes, such as the "recipe all" and "recipe possible" commands.
+
+Therefore, we need to standardize the way we parse
+the command in such a way that we do not have to worry about match priority or ordering, as we would if we
+were to implement this with simple if-else or switch statements.
+
+![RouterUML.png](images/RouterUML.png)
+
+- The "routing tree" is built around the concept of the `Routable` interface, with a `CommandRouterNode` at the root.
+- Classes which implement `Routable` have a `resolve` function which will map a `CommandTokens` object to an
+`ExecutableCommandFactory`.
+- Routing is done in a depth-first fashion via recursive calls to `resolve`, until an
+endpoint is reached at which point the corresponding `ExecutableCommandFactory` is returned.
+
+![RouterSnippetUML.png](images/RouterSnippetUML.png)
+
+This is an object diagram representing a portion of the routing tree for MealCompanion, supporting the "recipe all",
+"recipe possible", and "ingredient list" commands.
+
+The sequence diagram below illustrates the process for resolving the "recipe all" command.
+
+![RouterSequenceUML.png](images/RouterSequenceUML.png)
+
+### Ingredient Class
+
+Below shows the class diagram of how ingredients are being stored in our program
+
+![IngredientUML.png](images/IngredientUML.png)
```

Comment:
![Screenshot from 2023-03-31 13-15-51](https://user-images.githubusercontent.com/2891116/229029364-b6a2950b-a1c2-4a44-9768-85d59a6e595f.png)

There should not be a class circle in the UML diagram. This can be removed with `hide circle`. There should also not be red rectangles and green circles but rather minuses and plusses to indicate visibility.